qa-tests-backend/src/test/groovy/NetworkFlowTest.groovy

```diff
@@ -698,57 +697,81 @@ class NetworkFlowTest extends BaseSpecification {
-                log.info "${deploymentName} has incoming connections - ensure podSelectors/namespaceSelectors match " +
-                        "sources from graph"
-                def sourceDeploymentsFromGraph = outNodes.findAll { it.deploymentName }*.deploymentName
-                def sourceDeploymentsFromNetworkPolicy = ingressPodSelectors.collect {
-                    it."podSelector"."matchLabels"."app"
-                }
-                def sourceNamespacesFromNetworkPolicy = ingressNamespaceSelectors.collect {
-                    it."namespaceSelector"."matchLabels"."namespace.metadata.stackrox.io/name"
-                }.findAll { it != null }
-                sourceNamespacesFromNetworkPolicy.addAll(ingressNamespaceSelectors.collect {
-                    it."namespaceSelector"."matchLabels"."kubernetes.io/metadata.name"
-                }).findAll { it != null }
-                assert sourceDeploymentsFromNetworkPolicy.sort() == sourceDeploymentsFromGraph.sort()
-                if (!deployedNamespaces.containsAll(sourceNamespacesFromNetworkPolicy)) {
-                    log.info "Deployed namespaces do not contain all namespaces found in the network policy"
-                    log.info "The network policy:"
-                    log.info modification.toString()
+                if (allowAllIngress) {
+                    log.info "${deploymentName} has LB/External incoming traffic - ensure All Ingress allowed"
+                    assert yaml."spec"."ingress" == [[:]]
+                } else if (outNodes.size() > 0) {
+                    log.info "${deploymentName} has incoming connections - " +
+                        "ensure podSelectors/namespaceSelectors match sources from graph"
+                    def sourceDeploymentsFromGraph = outNodes.findAll { it.deploymentName }*.deploymentName
+                    def sourceDeploymentsFromNetworkPolicy = ingressPodSelectors.collect {
+                        it."podSelector"."matchLabels"."app"
+                    }
+                    def sourceNamespacesFromNetworkPolicy = ingressNamespaceSelectors.collect {
+                        it."namespaceSelector"."matchLabels"."namespace.metadata.stackrox.io/name"
+                    }.findAll { it != null }
+                    sourceNamespacesFromNetworkPolicy.addAll(ingressNamespaceSelectors.collect {
+                        it."namespaceSelector"."matchLabels"."kubernetes.io/metadata.name"
+                    }).findAll { it != null }
+
+                    log.debug("sourceDeploymentsFromNetworkPolicy: {}", sourceDeploymentsFromNetworkPolicy)
+                    log.debug("sourceDeploymentsFromGraph: {}", sourceDeploymentsFromGraph)
+                    // Additional assertions to define the desired state if this flakes in the future
+                    switch (deploymentName) {
+                        case TCPCONNECTIONTARGET:
+                            assert sourceDeploymentsFromNetworkPolicy.size() == 3
```

Comment:
Based on what I mentioned earlier, I think if the Network Policies are generated when the graph is not completed, this assertion will always fail no matter how many times we retry.